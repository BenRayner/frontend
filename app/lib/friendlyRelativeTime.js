import moment from 'moment';

const CALENDAR_LOCALE_STRINGS_TO_LOWERCASE = [
  'lastDay',
  'sameDay',
  'nextDay'
];

// Returns a friendly, relative version of a timestamp.
//
// For example:
//   "Today at 12:03 PM"
//   "Yesterday at 11:01 AM"
//   "Wed 13 Nov at 1:00 AM"
//   "Fri 1 Jan 2012 at 4:02 PM"
export default function friendlyRelativeTime(time, options = {}) {
  const formats = {
    sameElse: function(date) {return `ddd Do MMM${this.year() === moment(date).year() ? '' : ' YY'} [at] h:mm${options.seconds ? ':ss' : ''} A`;},
    lastWeek: function() {return `${options.inPast ? '[last] ' : ''}dddd [at] LT`;},
    nextWeek: function() {return `${options.inPast ? '' : '[next] '}dddd [at] LT`;}
  };

  if (options.capitalized !== true) {
    CALENDAR_LOCALE_STRINGS_TO_LOWERCASE.forEach((calendarString) => {
      formats[calendarString] = moment.localeData().calendar([calendarString]).replace(/\[[^\]]+\]/g, (replacement) => replacement.toLowerCase());
    });
  }

  return moment(time).calendar(moment(), formats);
}
