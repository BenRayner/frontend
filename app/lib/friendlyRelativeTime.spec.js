/* global describe, it, expect */
import MockDate from 'mockdate';
import friendlyRelativeTime from './friendlyRelativeTime';

const DATE_FIXTURES = [
  "2016-05-07T09:00:00.000Z",
  "2016-05-07T16:45:16.000Z",
  "2016-05-07T20:59:03.000Z",
  "2016-05-08T16:03:21.000Z",
  "2016-05-21T01:22:12.000Z",
  "2016-07-10T04:34:17.000Z",
  "2018-01-01T04:34:17.000Z"
];

const BOOL_FIXTURES = [
  undefined,
  true,
  false
];

const runDateFixtureSpecs = (options) => (() => {
  DATE_FIXTURES.forEach((date) => {
    it(`renders a correct time in the past or present`, () => {
      MockDate.set(date);
      expect(friendlyRelativeTime(DATE_FIXTURES[0], options)).toMatchSnapshot();
      MockDate.reset();
    });

    it(`renders a correct time in the future`, () => {
      MockDate.set(DATE_FIXTURES[0]);
      expect(friendlyRelativeTime(date, options)).toMatchSnapshot();
      MockDate.reset();
    });
  });
})

describe('friendlyRelativeTime', () => {
  BOOL_FIXTURES.forEach((seconds) => {
    BOOL_FIXTURES.forEach((capitalized) => {
      describe(`when supplied with options=\`{ seconds: ${seconds}, capitalized: ${capitalized} }\``, runDateFixtureSpecs({seconds, capitalized}));
    });
  });

  describe(`when supplied with options=undefined`, runDateFixtureSpecs());
});
