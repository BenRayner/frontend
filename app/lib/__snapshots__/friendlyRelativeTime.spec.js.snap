exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: false }\` renders a correct time in the future 1`] = `"today at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: false }\` renders a correct time in the future 2`] = `"Sun 8th May at 2:45 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: false }\` renders a correct time in the future 3`] = `"Sun 8th May at 6:59 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: false }\` renders a correct time in the future 4`] = `"Mon 9th May at 2:03 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: false }\` renders a correct time in the future 5`] = `"Sat 21st May at 11:22 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: false }\` renders a correct time in the future 6`] = `"Sun 10th Jul at 2:34 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: false }\` renders a correct time in the future 7`] = `"Mon 1st Jan 18 at 3:34 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: false }\` renders a correct time in the past or present 1`] = `"today at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: false }\` renders a correct time in the past or present 2`] = `"yesterday at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: false }\` renders a correct time in the past or present 3`] = `"yesterday at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: false }\` renders a correct time in the past or present 4`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: false }\` renders a correct time in the past or present 5`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: false }\` renders a correct time in the past or present 6`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: false }\` renders a correct time in the past or present 7`] = `"Sat 7th May 16 at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: true }\` renders a correct time in the future 1`] = `"Today at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: true }\` renders a correct time in the future 2`] = `"Sun 8th May at 2:45 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: true }\` renders a correct time in the future 3`] = `"Sun 8th May at 6:59 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: true }\` renders a correct time in the future 4`] = `"Mon 9th May at 2:03 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: true }\` renders a correct time in the future 5`] = `"Sat 21st May at 11:22 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: true }\` renders a correct time in the future 6`] = `"Sun 10th Jul at 2:34 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: true }\` renders a correct time in the future 7`] = `"Mon 1st Jan 18 at 3:34 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: true }\` renders a correct time in the past or present 1`] = `"Today at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: true }\` renders a correct time in the past or present 2`] = `"Yesterday at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: true }\` renders a correct time in the past or present 3`] = `"Yesterday at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: true }\` renders a correct time in the past or present 4`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: true }\` renders a correct time in the past or present 5`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: true }\` renders a correct time in the past or present 6`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: true }\` renders a correct time in the past or present 7`] = `"Sat 7th May 16 at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: undefined }\` renders a correct time in the future 1`] = `"today at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: undefined }\` renders a correct time in the future 2`] = `"Sun 8th May at 2:45 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: undefined }\` renders a correct time in the future 3`] = `"Sun 8th May at 6:59 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: undefined }\` renders a correct time in the future 4`] = `"Mon 9th May at 2:03 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: undefined }\` renders a correct time in the future 5`] = `"Sat 21st May at 11:22 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: undefined }\` renders a correct time in the future 6`] = `"Sun 10th Jul at 2:34 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: undefined }\` renders a correct time in the future 7`] = `"Mon 1st Jan 18 at 3:34 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: undefined }\` renders a correct time in the past or present 1`] = `"today at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: undefined }\` renders a correct time in the past or present 2`] = `"yesterday at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: undefined }\` renders a correct time in the past or present 3`] = `"yesterday at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: undefined }\` renders a correct time in the past or present 4`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: undefined }\` renders a correct time in the past or present 5`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: undefined }\` renders a correct time in the past or present 6`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: false, capitalized: undefined }\` renders a correct time in the past or present 7`] = `"Sat 7th May 16 at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: false }\` renders a correct time in the future 1`] = `"today at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: false }\` renders a correct time in the future 2`] = `"Sun 8th May at 2:45:16 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: false }\` renders a correct time in the future 3`] = `"Sun 8th May at 6:59:03 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: false }\` renders a correct time in the future 4`] = `"Mon 9th May at 2:03:21 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: false }\` renders a correct time in the future 5`] = `"Sat 21st May at 11:22:12 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: false }\` renders a correct time in the future 6`] = `"Sun 10th Jul at 2:34:17 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: false }\` renders a correct time in the future 7`] = `"Mon 1st Jan 18 at 3:34:17 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: false }\` renders a correct time in the past or present 1`] = `"today at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: false }\` renders a correct time in the past or present 2`] = `"yesterday at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: false }\` renders a correct time in the past or present 3`] = `"yesterday at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: false }\` renders a correct time in the past or present 4`] = `"Sat 7th May at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: false }\` renders a correct time in the past or present 5`] = `"Sat 7th May at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: false }\` renders a correct time in the past or present 6`] = `"Sat 7th May at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: false }\` renders a correct time in the past or present 7`] = `"Sat 7th May 16 at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: true }\` renders a correct time in the future 1`] = `"Today at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: true }\` renders a correct time in the future 2`] = `"Sun 8th May at 2:45:16 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: true }\` renders a correct time in the future 3`] = `"Sun 8th May at 6:59:03 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: true }\` renders a correct time in the future 4`] = `"Mon 9th May at 2:03:21 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: true }\` renders a correct time in the future 5`] = `"Sat 21st May at 11:22:12 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: true }\` renders a correct time in the future 6`] = `"Sun 10th Jul at 2:34:17 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: true }\` renders a correct time in the future 7`] = `"Mon 1st Jan 18 at 3:34:17 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: true }\` renders a correct time in the past or present 1`] = `"Today at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: true }\` renders a correct time in the past or present 2`] = `"Yesterday at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: true }\` renders a correct time in the past or present 3`] = `"Yesterday at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: true }\` renders a correct time in the past or present 4`] = `"Sat 7th May at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: true }\` renders a correct time in the past or present 5`] = `"Sat 7th May at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: true }\` renders a correct time in the past or present 6`] = `"Sat 7th May at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: true }\` renders a correct time in the past or present 7`] = `"Sat 7th May 16 at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: undefined }\` renders a correct time in the future 1`] = `"today at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: undefined }\` renders a correct time in the future 2`] = `"Sun 8th May at 2:45:16 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: undefined }\` renders a correct time in the future 3`] = `"Sun 8th May at 6:59:03 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: undefined }\` renders a correct time in the future 4`] = `"Mon 9th May at 2:03:21 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: undefined }\` renders a correct time in the future 5`] = `"Sat 21st May at 11:22:12 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: undefined }\` renders a correct time in the future 6`] = `"Sun 10th Jul at 2:34:17 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: undefined }\` renders a correct time in the future 7`] = `"Mon 1st Jan 18 at 3:34:17 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: undefined }\` renders a correct time in the past or present 1`] = `"today at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: undefined }\` renders a correct time in the past or present 2`] = `"yesterday at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: undefined }\` renders a correct time in the past or present 3`] = `"yesterday at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: undefined }\` renders a correct time in the past or present 4`] = `"Sat 7th May at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: undefined }\` renders a correct time in the past or present 5`] = `"Sat 7th May at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: undefined }\` renders a correct time in the past or present 6`] = `"Sat 7th May at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: true, capitalized: undefined }\` renders a correct time in the past or present 7`] = `"Sat 7th May 16 at 7:00:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: false }\` renders a correct time in the future 1`] = `"today at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: false }\` renders a correct time in the future 2`] = `"Sun 8th May at 2:45 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: false }\` renders a correct time in the future 3`] = `"Sun 8th May at 6:59 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: false }\` renders a correct time in the future 4`] = `"Mon 9th May at 2:03 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: false }\` renders a correct time in the future 5`] = `"Sat 21st May at 11:22 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: false }\` renders a correct time in the future 6`] = `"Sun 10th Jul at 2:34 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: false }\` renders a correct time in the future 7`] = `"Mon 1st Jan 18 at 3:34 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: false }\` renders a correct time in the past or present 1`] = `"today at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: false }\` renders a correct time in the past or present 2`] = `"yesterday at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: false }\` renders a correct time in the past or present 3`] = `"yesterday at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: false }\` renders a correct time in the past or present 4`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: false }\` renders a correct time in the past or present 5`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: false }\` renders a correct time in the past or present 6`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: false }\` renders a correct time in the past or present 7`] = `"Sat 7th May 16 at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: true }\` renders a correct time in the future 1`] = `"Today at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: true }\` renders a correct time in the future 2`] = `"Sun 8th May at 2:45 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: true }\` renders a correct time in the future 3`] = `"Sun 8th May at 6:59 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: true }\` renders a correct time in the future 4`] = `"Mon 9th May at 2:03 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: true }\` renders a correct time in the future 5`] = `"Sat 21st May at 11:22 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: true }\` renders a correct time in the future 6`] = `"Sun 10th Jul at 2:34 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: true }\` renders a correct time in the future 7`] = `"Mon 1st Jan 18 at 3:34 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: true }\` renders a correct time in the past or present 1`] = `"Today at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: true }\` renders a correct time in the past or present 2`] = `"Yesterday at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: true }\` renders a correct time in the past or present 3`] = `"Yesterday at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: true }\` renders a correct time in the past or present 4`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: true }\` renders a correct time in the past or present 5`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: true }\` renders a correct time in the past or present 6`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: true }\` renders a correct time in the past or present 7`] = `"Sat 7th May 16 at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: undefined }\` renders a correct time in the future 1`] = `"today at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: undefined }\` renders a correct time in the future 2`] = `"Sun 8th May at 2:45 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: undefined }\` renders a correct time in the future 3`] = `"Sun 8th May at 6:59 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: undefined }\` renders a correct time in the future 4`] = `"Mon 9th May at 2:03 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: undefined }\` renders a correct time in the future 5`] = `"Sat 21st May at 11:22 AM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: undefined }\` renders a correct time in the future 6`] = `"Sun 10th Jul at 2:34 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: undefined }\` renders a correct time in the future 7`] = `"Mon 1st Jan 18 at 3:34 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: undefined }\` renders a correct time in the past or present 1`] = `"today at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: undefined }\` renders a correct time in the past or present 2`] = `"yesterday at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: undefined }\` renders a correct time in the past or present 3`] = `"yesterday at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: undefined }\` renders a correct time in the past or present 4`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: undefined }\` renders a correct time in the past or present 5`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: undefined }\` renders a correct time in the past or present 6`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=\`{ seconds: undefined, capitalized: undefined }\` renders a correct time in the past or present 7`] = `"Sat 7th May 16 at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=undefined renders a correct time in the future 1`] = `"today at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=undefined renders a correct time in the future 2`] = `"Sun 8th May at 2:45 AM"`;

exports[`friendlyRelativeTime when supplied with options=undefined renders a correct time in the future 3`] = `"Sun 8th May at 6:59 AM"`;

exports[`friendlyRelativeTime when supplied with options=undefined renders a correct time in the future 4`] = `"Mon 9th May at 2:03 AM"`;

exports[`friendlyRelativeTime when supplied with options=undefined renders a correct time in the future 5`] = `"Sat 21st May at 11:22 AM"`;

exports[`friendlyRelativeTime when supplied with options=undefined renders a correct time in the future 6`] = `"Sun 10th Jul at 2:34 PM"`;

exports[`friendlyRelativeTime when supplied with options=undefined renders a correct time in the future 7`] = `"Mon 1st Jan 18 at 3:34 PM"`;

exports[`friendlyRelativeTime when supplied with options=undefined renders a correct time in the past or present 1`] = `"today at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=undefined renders a correct time in the past or present 2`] = `"yesterday at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=undefined renders a correct time in the past or present 3`] = `"yesterday at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=undefined renders a correct time in the past or present 4`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=undefined renders a correct time in the past or present 5`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=undefined renders a correct time in the past or present 6`] = `"Sat 7th May at 7:00 PM"`;

exports[`friendlyRelativeTime when supplied with options=undefined renders a correct time in the past or present 7`] = `"Sat 7th May 16 at 7:00 PM"`;
